In [15]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [16]:
import sys
sys.path.append("./../")

In [25]:
import time
import cProfile
import re
import numpy as np
from datasets.multimnist import MultiMNist

In [18]:
root='/mnt/data/datasets/multimnist_test'
cProfile.run('MultiMNist(root=root,train=True, generate=True, g_samples=[10,10])')

generate multimnist...
torch.Size([10, 28, 28])
torch.Size([10, 28, 28])
generation done
         131121 function calls (128521 primitive calls) in 3.967 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
       24    0.000    0.000    0.001    0.000 <__array_function__ internals>:2(prod)
       20    0.000    0.000    0.002    0.000 <__array_function__ internals>:2(where)
        1    0.006    0.006    3.967    3.967 <string>:1(<module>)
      200    0.002    0.000    0.078    0.000 Image.py:2158(save)
      600    0.001    0.000    0.001    0.000 Image.py:2651(_check_size)
      200    0.001    0.000    0.003    0.000 Image.py:2669(new)
      200    0.001    0.000    0.006    0.000 Image.py:2708(frombytes)
      200    0.000    0.000    0.007    0.000 Image.py:2746(frombuffer)
      200    0.003    0.000    0.017    0.000 Image.py:2799(fromarray)
      200    0.000    0.000    0.000    0.000 Image.py:341(preinit)
      200  

In [30]:
start = time.time()

a = np.arange(0,1000,1)

for i in a:
    np.square(a[i])

end = time.time()

print("{:.5f}".format(end - start))

0.00145


In [33]:
start = time.time()

MultiMNist(root='/mnt/data/datasets/multimnist_test',train=True, generate=True, g_samples=[10,10])

end = time.time()

print("{:.5f}".format(end - start))

generate multimnist...
torch.Size([600, 28, 28])
torch.Size([600, 28, 28])
generation done
133.44664


samples 1200 gen 10 time 133.44664 s => 89.923 Img/ sec => 0.01112 s/Img -> Time Full Dataset 9 days
    -> that wrong...


In [50]:
a = end - start
samplesize = 12000
b = a / samplesize 
c = 1/ b
d = (b * 70000000)
e = d / (60*60*24)


print(a)
print(b)
print(c)
print(d)
print(e)

133.44664478302002
0.011120553731918335
89.92357971616009
778438.7612342834
9.00970788465606
